shell/platform/windows/client_wrapper/include/flutter/flutter_window_controller.h

```diff
@@ -0,0 +1,69 @@
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_CLIENT_WRAPPER_INCLUDE_FLUTTER_FLUTTER_WINDOW_CONTROLLER_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_CLIENT_WRAPPER_INCLUDE_FLUTTER_FLUTTER_WINDOW_CONTROLLER_H_
+
+#include <mutex>
+
+#include "flutter_engine.h"
+#include "method_channel.h"
+#include "win32_wrapper.h"
+#include "windowing.h"
+
+namespace flutter {
+
+// A singleton controller for Flutter windows.
```

Comment:
Could you expand why this needs to be a singleton? What would I do if my app has two Flutter engines?